File: app/src/main/java/com/mxt/anitrend/adapter/pager/detail/ProfilePageAdapter.java
package com.mxt.anitrend.adapter.pager.detail;

import android.content.Context;

import androidx.fragment.app.Fragment;
import androidx.fragment.app.FragmentManager;

import com.mxt.anitrend.R;
import com.mxt.anitrend.base.custom.pager.BaseStatePageAdapter;
import com.mxt.anitrend.util.GraphUtil;
import com.mxt.anitrend.util.KeyUtil;
import com.mxt.anitrend.view.fragment.detail.UserFeedFragment;
import com.mxt.anitrend.view.fragment.detail.UserOverviewFragment;

/**
 * Created by max on 2017/11/16.
 */

public class ProfilePageAdapter extends BaseStatePageAdapter {

    public ProfilePageAdapter(FragmentManager fragmentManager, Context context) {
        super(fragmentManager, context);
        setPagerTitles(R.array.profile_page_titles);
    }

    /**
     * Return the Fragment associated with a specified position.
     *
     * @param position
     */
    @Override
    public Fragment getItem(int position) {
        switch (position) {
            case 0:
                return UserOverviewFragment.newInstance(getParams());
            case 1:
                return UserFeedFragment.newInstance(getParams(), GraphUtil.INSTANCE.getDefaultQuery(true)
                        .putVariable(KeyUtil.arg_type, KeyUtil.MEDIA_LIST));
            case 2:
                return UserFeedFragment.newInstance(getParams(), GraphUtil.INSTANCE.getDefaultQuery(true)
                        .putVariable(KeyUtil.arg_type, KeyUtil.TEXT));
        }
        return null;
    }
}

File: app/src/main/java/com/mxt/anitrend/adapter/pager/detail/ProfilePageAdapter.kt
package com.mxt.anitrend.adapter.pager.detail

import android.content.Context
import android.os.Bundle
import androidx.fragment.app.Fragment
import androidx.fragment.app.FragmentActivity
import com.mxt.anitrend.R
import com.mxt.anitrend.base.custom.pager.BaseStatePageAdapter
import com.mxt.anitrend.extension.LAZY_MODE_UNSAFE
import com.mxt.anitrend.ui.fragmentByTagOrNew
import com.mxt.anitrend.ui.model.FragmentItem
import com.mxt.anitrend.util.KeyUtil
import com.mxt.anitrend.view.fragment.detail.UserFeedFragment
import com.mxt.anitrend.view.fragment.detail.UserOverviewFragment

/**
 * Created by max on 2017/11/16.
 */
class ProfilePageAdapter(
    fragmentActivity: FragmentActivity,
    context: Context,
) : BaseStatePageAdapter(fragmentActivity, context) {
    init {
        setPagerTitles(R.array.profile_page_titles)
    }

    private val fragmentItems by lazy(LAZY_MODE_UNSAFE) {
        listOf<FragmentItem<Fragment>>(
            FragmentItem(UserOverviewFragment::class.java, Bundle(params)),
            FragmentItem(
                UserFeedFragment::class.java,
                Bundle(params).apply {
                    putString(KeyUtil.arg_type, KeyUtil.MEDIA_LIST)
                    putInt(KeyUtil.arg_page_limit, KeyUtil.PAGING_LIMIT)
                },
                "UserFeedFragmentMediaList",
            ),
            FragmentItem(
                UserFeedFragment::class.java,
                Bundle(params).apply {
                    putString(KeyUtil.arg_type, KeyUtil.TEXT)
                    putInt(KeyUtil.arg_page_limit, KeyUtil.PAGING_LIMIT)
                },
                "UserFeedFragmentText",
            ),
        )
    }

    override fun createFragment(position: Int): Fragment = fragmentItems[position].fragmentByTagOrNew(fragmentActivity)
}
